controllers/farmer/updateFarmer.js

```diff
@@ -0,0 +1,62 @@
+const { models } = require('../../models');
+const {
+  createError,
+  GENERIC_ERROR,
+  NOT_FOUND
+} = require('../../helpers/error');
+
+/**
+ * @description Update farmer details
+ *
+ * @param {object} req
+ * @param {object} res
+ * @param {object} next
+ */
+const updateFarmer = async (req, res, next) => {
+  try {
+    const farmerId = req.params.id;
+    const farmerDetails = req.body;
+    const { username } = req.user;
+
+    const farmerExist = await models.Farmer.findOne({ _id: farmerId });
+    if (!farmerExist) {
+      return next(
+        createError({
+          message: 'Farmer does not exist',
+          status: NOT_FOUND
+        })
+      );
+    }
+    if (farmerExist.staff !== username) {
```

Comment:
This is a great check. `Admin` should have the ability to update the user as well tho. (In case the `staff` member that was managing this farmer is not available)